src/session/session_helper.c

```diff
@@ -134,3 +134,35 @@ __wt_session_dump(WT_SESSION_IMPL *session, WT_SESSION_IMPL *dump_session, bool
     __wt_scr_free(session, &buf);
     return (ret);
 }
+
+/*
+ * __wt_session_set_last_error --
+ *     Stores information about the last error to occur during this session.
+ */
+int
+__wt_session_set_last_error(
+  WT_SESSION *wt_session, int err, int sub_level_err, const char *err_msg_content)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    size_t err_msg_size;
+
+    /* Ensure arguments are valid, and that session is not a mock */
+    WT_ASSERT(session, __wt_is_valid_sub_level_error(sub_level_err));
+    if (err_msg_content == NULL || wt_session->get_last_error == NULL)
+        return (0);
+
+    /* Free the last error message string, if it was allocated. */
+    __wt_free(session, err_msg);
+
+    /* Load error codes and message content into err_info. */
+    err_msg_size = (strlen(err_msg_content) + 1) * sizeof(char);
+    WT_ERR(__wt_malloc(session, err_msg_size, &err_msg));
+    WT_ERR(__wt_snprintf(err_msg, err_msg_size, "%s", err_msg_content));
+    session->err_info = (WT_ERROR_INFO){err, sub_level_err, err_msg};
```

Comment:
I'm not a fan of creating anonymous initializers. In this case it may be dangerous. We are pointing err_info pointer  to memory location that has been created in a static memory.

Reply:
So is it better to reassign `err`/`sub_level_err`/`err_msg` one by one instead?

Reply:
I agree with that, this code is a bit more error prone. E.G. you reorder the fields in `WT_ERROR_INFO` struct or inserted some new fields in it, and you forget to change the anonymous initializers, which causes a bug.

Reply:
Done in [Updated mock session and altered __wt_session_set_last_error](https://github.com/wiredtiger/wiredtiger/pull/11361/commits/6c29c7c8e3950f9ff8d5b27c593f29f8bf857afc)

Reply:
Thanks this change looks good to me, since Jie opened this conversation I will leave him to resolve it.